### DISCUSSION TITLE
forecast.solar kein API key möglich

### DISCUSSION THREAD

**Discussion by thorsten-gehrig** [wiggal/GEN24_Ladesteuerung#63] [General]
Hallo
wenn ich das recht sehe, ist bei forecast.solar es nicht möglich den API key einzugeben.
Wenn ich es recht verstehe, würde man mit API key die Daten mit 30min Auflösung bekommen - free "nur" 60 Minuten.
Rate limit "anonym"=12 pro Tag, rate limit personal =60 pro Tag
Die URL für die Nutzung mit API ist nur minimal angepasst... ist vermutlich einfach in WeatherDataProvider2.py anzupassen...

Könnte man das mit Einbauen? Außerdem wäre es schön die APIs nur in einem bestimmten Zeitfenster abzufragen - um die limits zu schonen (zB von 6-20 uhr)

Gruß
Thorsten

**Comment by wiggal** [comment]
Hallo Thorsten,

die API in einer 30min Auflösung abzufragen bringt in dem Fall nichts, da die ganze Prognoseberechnung auf stündliche Prognosewerte aufgebaut ist. Von solcast.com enthält die API sogar eine  30min Auflösung, die aber beim Aufbereiten auf eine Stundenauflösung reduziert wird.

Wann und wie oft die API abgefragt wird, steuerst du mit dem crontab Eintrag, einfach hier anpassen.
`33 5,8,10,12,14,19 * * * /DIR/start_PythonScript.sh WeatherDataProvider2.py`

Ich hoffe die Antwort ist befriedigend
Schönen Gruß
